### PR TITLE
feat(protocol): user smaller cooldown windows

### PR DESCRIPTION
**Pull request by dantaik**
Use smaller cooldown windows:

- changed SGX/ZK cooldown window from 1 day to 4 hours
- changed SGX+ZK cooldown window from 1 day to 2 hours
- changed guardian prover cooldown window to 4 hours

Also changed guardian provers' proving window to 0 (as this value is not used at all)

### CODE REVIEW

File: packages/protocol/contracts/layer1/based/LibProving.sol [hunk 1]
```diff
@@ -326,8 +327,8 @@ library LibProving {
         // The assigned prover is granted exclusive permission to prove only the first
         // transition.
         if (
-            local.tier.contestBond != 0 && ts.contester == address(0) && local.tid == 1
-                && ts.tier == 0 && local.inProvingWindow
+            !local.isTopTier && ts.contester == address(0) && local.tid == 1 && ts.tier == 0
```
Comment (dantaik) [inline packages/protocol/contracts/layer1/based/LibProving.sol:330]: no logical changes, only to reuse "local.isTopTieer"

File: packages/protocol/contracts/layer1/tiers/ITierProvider.sol [hunk 2]
```diff
@@ -21,12 +21,6 @@ interface ITierProvider {
     /// @return Tier struct containing the tier's parameters.
     function getTier(uint16 tierId) external view returns (Tier memory);
 
-    /// @dev Retrieves the IDs of all supported tiers.
-    /// Note that the core protocol requires the number of tiers to be smaller
-    /// than 256. In reality, this number should be much smaller.
-    /// @return The ids of the tiers.
-    function getTierIds() external view returns (uint16[] memory);
```
Comment (YoGhurt111) [inline packages/protocol/contracts/layer1/tiers/ITierProvider.sol:28]: The client needs to adapt to this change.

File: packages/protocol/contracts/layer1/tiers/TierProviderBase.sol [hunk 3]
```diff
@@ -16,12 +16,27 @@ abstract contract TierProviderBase is ITierProvider {
     uint16 public constant GRACE_PERIOD = 240; // minutes
     uint96 public constant BOND_UNIT = 75 ether; // TAIKO tokens
 
+    /// @inheritdoc ITierProvider
+    function getTierIds() external pure returns (uint16[] memory ids_) {
```
Comment (YoGhurt111) [inline packages/protocol/contracts/layer1/tiers/TierProviderBase.sol:20]: Still need some minor changes to adapt these not-yet-supported tiers. 
Let me do it.

Comment (dantaik) [reply packages/protocol/contracts/layer1/tiers/TierProviderBase.sol:20]: No worries, I can change it my way.

Comment (YoGhurt111) [reply packages/protocol/contracts/layer1/tiers/TierProviderBase.sol:20]: I already submitted pr https://github.com/taikoxyz/taiko-mono/pull/18349/files to adapt this change.

**Review by YoGhurt111 — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**